tests/plugins/test_ddp_fully_sharded_with_full_state_dict.py

```diff
@@ -0,0 +1,188 @@
+import os
+from typing import Any, Dict, Optional
+from unittest import mock
+
+import pytest
+import torch
+
+from pytorch_lightning import Trainer
+from pytorch_lightning.callbacks import ModelCheckpoint
+from pytorch_lightning.plugins import DDPFullyShardedPlugin, FullyShardedNativeMixedPrecisionPlugin
+from pytorch_lightning.utilities import _FAIRSCALE_FULLY_SHARDED_AVAILABLE
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from tests.helpers.boring_model import BoringModel
+from tests.helpers.runif import RunIf
+
+if _FAIRSCALE_FULLY_SHARDED_AVAILABLE:
+    from fairscale.nn import FullyShardedDataParallel, wrap
+
+
+@RunIf(fairscale_fully_sharded=True)
+def test_sharded_ddp_choice(tmpdir):
+    """
+    Test to ensure that plugin is correctly chosen
+    """
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        fast_dev_run=True,
+        plugins="fsdp",
+    )
+    assert isinstance(trainer.accelerator.training_type_plugin, DDPFullyShardedPlugin)
+
```

Comment:
I thought it's not supported on CPU?
Can't we evaluate compatibility in the accelerator connector?

Reply:
nice catch here, it passed as our assertion for GPU happens in `setup_distributed`, not when initialize Trainer. I think we could do it in accelerator connector, but personal feeling is that file is becoming too giant and pretty complex, try to avoid additional logic there, feel it is specific plugin/accelerator strategy's responsibility to check when environment is setup.